pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -191,10 +191,12 @@ func (builder *nodeEntityDTOBuilder) getNodeCommoditiesSold(node *api.Node, clus
 	if err != nil {
 		return nil, true, fmt.Errorf("failed to get cpu frequency from sink for node %s: %s", key, err)
 	}
-	// cpu and cpu request needs to be converted from number of cores to frequency.
+	// cpu and cpu request needs to be converted from number of millicores to frequency.
 	converter := NewConverter().Set(
 		func(input float64) float64 {
-			return input * cpuFrequency
+			// All cpu metrics are stored in millicores in metrics sink for consistency
+			// But we send the node cpu metrics in MHz.
```

Comment:
Since all cpu metrics are stored in millicores for the node, we need to update the logic in line 142 of the same file as well to convert numCPUs from "millicores" to "cores" and send to the platform:
```
		cpuMetricValue, err := builder.metricValue(metrics.NodeType, nodeKey, metrics.CPU, metrics.Capacity, nil)
		if err != nil {
			glog.Errorf("Failed to get number of CPU in cores for VM %s: %v", nodeKey, err)
			continue
		}
		cpuCores := int32(cpuMetricValue.Avg)
```
to something like:
```
                 cpuMetricValue, err := builder.metricValue(metrics.NodeType, nodeKey, metrics.CPU, metrics.Capacity, nil)
		if err != nil {
			glog.Errorf("Failed to get number of CPU in millicores for VM %s: %v", nodeKey, err)
			continue
		}
		cpuCores := int32(util.MetricMilliToUnit(cpuMetricValue.Avg))
```
`numCPUs` field is treated as CPUs in "cores" in platform side discovered from different probes and will be used to calculate CPU speed (cc @ading1977). I think we want to make this field consistent with other probes so that stitching won't be affected.

Reply:
Updated!